File: laprak1/python/mahasiswa.py
from typing import List


class Mahasiswa:
    def __init__(self, Nama="", Nim=0, Prodi="-", Fakultas="-"):
        self.Nama = Nama
        self.Nim = Nim
        self.Prodi = Prodi
        self.Fakultas = Fakultas

    def get_Nama(self):
        return self.Nama

    def set_Nama(self, Nama):
        self.Nama = Nama

    def get_Nim(self):
        return self.Nim

    def set_Nim(self, Nim):
        self.Nim = Nim

    def get_Prodi(self):
        return self.Prodi

    def set_Prodi(self, Prodi):
        self.Prodi = Prodi

    def get_Fakultas(self):
        return self.Fakultas

    def set_Fakultas(self, Fakultas):
        self.Fakultas = Fakultas

    def tambah(self, n: List['Mahasiswa'], temp: 'Mahasiswa'):
        n.append(temp)

    def kurang(self, n: List['Mahasiswa'], temp: 'Mahasiswa'):
        flag = 0
        ada = 0
        for x, i in enumerate(n):
            if i.get_Nim() == temp.get_Nim():
                ada = 1
                flag = x

        if ada == 1:
            n.pop(flag)
        else:
            print("NIM tidak ditemukan")

    def ubah(self, n: List['Mahasiswa'], temp: 'Mahasiswa'):
        flag = 0
        ada = 0
        for x, i in enumerate(n):
            if i.get_Nim() == temp.get_Nim():
                ada = 1
                flag = x
            else:
                ada = 2

        if ada == 1:
            n.insert(flag, temp)
            n.pop(flag+1)
        else:
            print("NIM yang dicari tidak ditemukan")

File: laprak1/cpp/mahasiswa.cpp
#include <iostream>
#include <string>
#include <list>
#include <iterator>

using namespace std;

class mahasiswa
{
private:
    string Nama;
    int Nim;
    string Prodi;
    string Fakultas;

public:
    mahasiswa()
    {
        this->Nama = "";
        this->Nim = 0;
        this->Prodi = "-";
        this->Fakultas = "-";
    };

    mahasiswa(string Nama, int Nim, string Prodi, string Fakultas)
    {
        this->Nama = Nama;
        this->Nim = Nim;
        this->Prodi = Prodi;
        this->Fakultas = Fakultas;
    }

    string get_Nama()
    {
        return this->Nama;
    }

    void set_Nama(string Nama)
    {
        this->Nama = Nama;
    }

    int get_Nim()
    {
        return this->Nim;
    }

    void set_Nim(int Nim)
    {
        this->Nim = Nim;
    }

    string get_Prodi()
    {
        return this->Prodi;
    }

    void set_Prodi(string Prodi)
    {
        this->Prodi = Prodi;
    }

    string get_Fakultas()
    {
        return this->Fakultas;
    }

    void set_Fakultas(string Fakultas)
    {
        this->Fakultas = Fakultas;
    }

    void tambah(list<mahasiswa> &n, mahasiswa temp)
    {
        n.push_back(temp);
    }

    void kurang(list<mahasiswa> &n, mahasiswa temp)
    {
        list<mahasiswa>::iterator tmp;
        int x = 0;
        int flag = 0;
        int ada = 0;
        for (list<mahasiswa>::iterator i = n.begin(); i != n.end(); i++)
        {
            if (i->get_Nim() == temp.get_Nim())
            {
                ada = 1;
                flag = x;
            }
            x++;
        }

        if (ada == 1)
        {
            tmp = n.begin();
            for (int j = 0; j < flag; j++)
            {
                ++tmp;
            }
            n.erase(tmp);
        }
        else
        {
            cout << "NIM tidak ditemukan\n";
        }
    }

    void ubah(list<mahasiswa> &n, mahasiswa temp)
    {
        int flag = 0;
        int ada = 0;
        int x = 0;
        for (list<mahasiswa>::iterator i = n.begin(); i != n.end(); i++)
        {
            if (i->get_Nim() == temp.get_Nim())
            {
                ada = 1;
                flag = x;
            }
            x++;
        }

        if (ada == 1)
        {
            list<mahasiswa>::iterator tmp;
            tmp = n.begin();
            for (int j = 0; j < flag; j++)
            {
                ++tmp;
            }
            n.insert(tmp, temp);
            n.erase(tmp);
        }
        else
        {
            cout << "NIM yang dicari tidak ditemukan\n\n";
        }
    }

    ~mahasiswa()
    {
    }
};
